src/wp-admin/includes/class-wp-comments-list-table.php

```diff
@@ -663,7 +664,7 @@ public function single_row( $item ) {
 	 *                or if the current user cannot edit the comment.
 	 */
 	protected function handle_row_actions( $item, $column_name, $primary ) {
-		global $comment_status;
+		global $comment_status, $hook_suffix;
```

Comment:
This change is needed here (or in `wp_comment_reply()`, or elsewhere in the trace) to resolve [test failures](https://github.com/WordPress/wordpress-develop/runs/8182974480?check_suite_focus=true). While this change could also be applied in the tests, this is a singular change in source that will ensure users don't also encounter the same issue in unforeseen cases.

However, from what myself and @mukeshpanchal27 have been able to determine, this is part of a larger issue where `$id = $GLOBALS['hook_suffix']` [here](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-admin/includes/class-wp-screen.php#L223) is not guarded, and assumes that `WP_Screen::get()` will only be called in a context where the global is set.

There have been [several tickets](https://core.trac.wordpress.org/search?q=undefined+index+hook_suffix+ajax&noquickjump=1&ticket=on) on Trac raising this issue when using AJAX, most notably (IMO), [this ticket](https://core.trac.wordpress.org/ticket/49089), which includes [PR 1490](https://github.com/WordPress/wordpress-develop/pull/1490). Manually applying the changes from PR 1490 resolves the test failures in this PR. Both myself and @mukeshpanchal27 have run the AJAX tests and full test suite (Single site + Multisite) and all tests pass.

I think it's worth considering PR 1490 for merging to `trunk`, as this will have a much wider positive impact, and the changes look safe. Pinging @getsource, @dream-encode and @jeffpaul as 6.1 Core tech leads for opinions on this.

Adding the `$hook_suffix` global has been kept in the PR in case this is the preferred approach after feedback.